layers/radiotap.go

```diff
@@ -859,12 +859,20 @@ func (m *RadioTap) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) erro
 	}
 
 	payload := data[m.Length:]
-	if !m.Flags.FCS() { // Dot11.DecodeFromBytes() expects FCS present
-		fcs := make([]byte, 4)
+	if !m.Flags.FCS() {
+		// Dot11.DecodeFromBytes() expects FCS present and performs a hard chop on the checksum
+		// If a user is handing in subslices or packets from a buffered stream, the capacity of the slice
+		// may extend beyond the len, rather than expecting callers to enforce cap==len on every packet
+		// we take the hit in this one case and do a reallocation.  If the user DOES enforce cap==len
+		// then the reallocation will happen anyway on the append.  This is requried because the append
+		// write to the memory directly after the payload if there is sufficient capacity, which callers
+		// may not expect.
+		reallocPayload := make([]byte, len(payload)+4)
+		copy(reallocPayload[0:len(payload)], payload)
```

Comment:
one brief note:  copy(x, y) will copy until it runs out of one buffer (IE: it'll copy bytes from A->B until either A or B runs out), so there actually isn't a need to slice the reallocPayload here.  Certainly doesn't hurt, though :)